react-main/src/app/components/WithLoading/index.js

```diff
@@ -0,0 +1,23 @@
+import React, { Fragment } from 'react';
+import { RingLoader } from 'react-spinners';
+import styles from './styles.scss';
+
+function WithLoading(WrappedComponent) {
+  return function WihLoadingComponent({ isLoading, ...props }) {
+    return (
+      <Fragment>
+        <WrappedComponent {...props} />
```

Comment:
Isn't the screen visible when the spinner is working? I think the user will be able to see and interact with the screen when loading, and it shouldn't.